nexus/src/app/background/init.rs

```diff
@@ -39,128 +122,208 @@ use oximeter::types::ProducerRegistry;
+        };
 
-    /// task handle for propagation of VPC router rules to all OPTE ports
-    pub task_vpc_route_manager: TaskHandle,
-}
+        (initializer, background_tasks)
+    }
 
-impl BackgroundTasks {
-    /// Kick off all background tasks
+    /// Starts all the Nexus background tasks
+    ///
+    /// This function will wire up the `Activator`s in `background_tasks` to the
+    /// corresponding tasks once they've been started.
     #[allow(clippy::too_many_arguments)]
     pub fn start(
-        opctx: &OpContext,
+        self,
+        background_tasks: &'_ BackgroundTasks,
+        opctx: OpContext,
         datastore: Arc<DataStore>,
-        config: &BackgroundTaskConfig,
+        config: BackgroundTaskConfig,
         rack_id: Uuid,
         nexus_id: Uuid,
         resolver: internal_dns::resolver::Resolver,
         saga_request: Sender<SagaRequest>,
-        v2p_watcher: (
-            tokio::sync::watch::Sender<()>,
-            tokio::sync::watch::Receiver<()>,
-        ),
-        producer_registry: &ProducerRegistry,
-    ) -> BackgroundTasks {
-        let mut driver = Driver::new();
+        v2p_watcher: (watch::Sender<()>, watch::Receiver<()>),
+        producer_registry: ProducerRegistry,
+    ) -> Driver {
+        let mut driver = self.driver;
+        let opctx = &opctx;
+        let producer_registry = &producer_registry;
+
+        // This "let" construction helps catch mistakes where someone forgets to
+        // wire up an activator to its corresponding background task.
+        let BackgroundTasks {
+            task_internal_dns_config,
+            task_internal_dns_servers,
+            task_internal_dns_propagation,
+            task_external_dns_config,
+            task_external_dns_servers,
+            task_external_dns_propagation,
+            task_metrics_producer_gc,
+            task_external_endpoints,
+            task_nat_cleanup,
+            task_bfd_manager,
+            task_inventory_collection,
+            task_physical_disk_adoption,
+            task_phantom_disks,
+            task_blueprint_loader,
+            task_blueprint_executor,
+            task_crdb_node_id_collector,
+            task_service_zone_nat_tracker,
+            task_switch_port_settings_manager,
+            task_v2p_manager,
+            task_region_replacement,
+            task_region_replacement_driver,
+            task_instance_watcher,
+            task_service_firewall_propagation,
+            task_abandoned_vmm_reaper,
+            task_vpc_route_manager,
+            // Add new background tasks here.  Be sure to use this binding in a
+            // call to `Driver::register()` below.  That's what actually wires
+            // up the Activator to the corresponding background task.
+
+            // The following fields can be safely ignored here because they're
+            // already wired up as needed.
+            external_endpoints: _external_endpoints,
```

Comment:
nit, take it or leave it: i _think_ this can just be an `_` since we don't need the binding to live until the function returns?
```suggestion
            external_endpoints: _,
```